src/v/datalake/translation/translation_probe.cc

```diff
@@ -36,6 +36,53 @@ translation_probe::translation_probe(model::ntp ntp)
 void translation_probe::register_invalid_record_metric() {
     namespace sm = ss::metrics;
 
+    std::vector<sm::label_instance> ntp_labels{
+      namespace_label(_ntp.ns()),
+      topic_label(_ntp.tp.topic()),
+      partition_label(_ntp.tp.partition()),
+    };
+
+    _public_metrics->add_group(
+      group_name,
+      {
+        sm::make_counter(
+          "translations_finished",
+          _translations_finished,
+          sm::description("Number of finished translator executions"),
+          ntp_labels)
+          .aggregate({
+            sm::shard_label,
+            partition_label,
+          }),
+        sm::make_counter(
+          "files_created",
+          _files_created,
+          sm::description("Number of created parquet files"),
+          ntp_labels)
+          .aggregate({
+            sm::shard_label,
+            partition_label,
+          }),
+        sm::make_counter(
+          "rows_added",
+          _rows_added,
+          sm::description("Number of rows in created parquet files"),
+          ntp_labels)
+          .aggregate({
+            sm::shard_label,
+            partition_label,
+          }),
+        sm::make_counter(
+          "bytes_added",
+          _bytes_added,
+          sm::description("Number of bytes in created parquet files"),
+          ntp_labels)
+          .aggregate({
+            sm::shard_label,
+            partition_label,
+          }),
```

Comment:
nit: might be worth prefixing each of these with `parquet_` to disambiguate between e.g. kafka bytes